index.bs

```diff
@@ -345,19 +345,21 @@ or failure:
 1. <a>Parse a list of component values</a> |marginString|,
 	storing the result as |tokens|.
 2. Remove all whitespace tokens from |tokens|.
-3. If the length of |tokens| is 0 or greater than 4,
+3. If the length of |tokens| is greater than 4,
 	return failure.
 4. Replace each |token| in |tokens|:
 	* If |token| is an <a>absolute length</a> <a>dimension</a> token,
 		replace it with a an equivalent pixel length.
 	* If |token| is a <<percentage>> token,
 		replace it with an equivalent percentage.
 	* Otherwise, return failure.
-5. If there is one element in |tokens|,
+5. If there are zero elements in |tokens|,
+	set |tokens| to ["0", "0", "0", "0"].
```

Comment:
This should be done before parsing, right? Otherwise you probably need to word it as "an absolute lengths of zero pixels" or something.

Also, this doesn't match Blink, as far as I can tell. Per this change, `new IntersectionObserver(() => {}, { rootMargin: " " })` should end up with a zero margin, while it seems like Blink special-cases the empty string.

Reply:
Before parsing also doesn't work, because 0 is a number token, not a dimension or percentage token.

Reply:
I changed it to apply the default value before parsing. I also added WPT coverage, and fixed chromium's behavior:

https://chromium-review.googlesource.com/c/chromium/src/+/2388706/1/third_party/blink/web_tests/external/wpt/intersection-observer/observer-attributes.html